draft-ietf-sidrops-publication-server-bcp-01.md

```diff
@@ -92,7 +92,29 @@ Protocol. The Publication Server generates the content for the public-facing
 RRDP and Rsync Repositories. It is strongly RECOMMENDED that these functions
 are separated from serving the repository content.
 
-## Availability
+## Self Hosted Publication Server
+
+Some organisations that use a self-hosted CA, rather than for example a hosted
+CA as service provided by their RIR or NIR, also run a self-hosted Publication
+Server for their CA. In this case, the organisation is responsible for ensuring
+the availability of the RRDP and rsync content as described in section 5 and 6
+of this document.
+
+Because RPs use cached data, short outages don't need to cause immediate issues
+if these organisations fix their repositories before objects expire and ensure
+that their Publication Server ([@!RFC8181]) is available when there is a need to
+update RPKI objects such as ROAs.
+
+However, availability issues with such repositories are frequent and negatively
+impact RPs. Therefore, it is strongly RECOMMENDED that organistions use a
+publication service provided by their RIR, NIR or other parent as much as
+possible.
+
+Furthermore, it is RECOMMENDED that CAs that act as a parent and which operate
+a Publication Server offer publication as a service to CAs under their CA
+hierarchy, i.e. direct child CAs, grandchildren, great grandchildren, etc.
```

Comment:
RECOMMENDED for direct child CAs makes sense, but offering publication past that point in situations where the CA doesn't have a separate business relationship with the grandchild and where publication is not mediated by the direct child (e.g. via a proxy) is not always (or even often) going to be a good idea, due e.g. to having to provide direct technical support to those grandchild CAs.

Reply:
Ok, mea culpa, this was sloppy phrasing.

I think that direct child CAs are a minimum we can agree on. So perhaps the text should just say that.

But at RIPE NCC we let our members configure up to 10 (possibly more in future) publishers when they run a delegated CA. The member remains responsible for these the publishers, the XML exchange, and supporting them. That is kind of what I wanted to capture here as one of the main use cases for running a self-hosted CA in the first place is that ability to create delegated CAs and if that means multiple repositories then that is not great.

But as you mentioned, the child CA can operate a publication proxy and this would work even if the parent only offers publication to the direct child. I know that APNIC has an implementation [here](https://github.com/APNIC-net/rpki-publication-proxy). Is that something we can/should reference here? When I worked on Krill I thought of implementing it there as well, but never got to it.

In any case, let me try to rephrase this. How about:

```
Furthermore, it is RECOMMENDED that CAs that act as a parent and which operate
a Publication Server offer publication as a service to their children. They MAY
also allow their children to register additional publishers, e.g. for their
delegated child CAs, i.e. grandchildren of the aforementioned parent. In this
case the child CA organisation would be responsible for the [@!RFC8183] identity
exchange and support for the additional publishers. On the other hand, child CAs
can also act as an [@!RFC8181] compliant [@rpki-publication-proxy] and publish
on behalf of their children.
```

Comments welcome.

Reply:
Yep, I think that text is closer, but it has knock-on effects on the previous paragraph, and some comments about referrals might be useful as well.  Suggested updates:

```
However, availability issues with such repositories are frequent and negatively impact RPs, and the greater the number of separate repositories, the greater the chance of such problems.  Therefore, CAs that act as parents of other CAs are RECOMMENDED to provide a publication service for their child CAs, and CAs with a parent who offers a publication service are RECOMMENDED to use that service, instead of running their own.

For the case of a 'grandchild' CA, where CA1 is a TA, CA2 is a child CA of CA1, and CA3 is a child CA of CA2, there are several options for providing publication service to CA3:

 1. RFC 8183 defines a 'referral' mechanism as part of the out-of-band CA setup protocol.  If supported by CA1 and CA2, then this simplifies the process of registering CA3 as a direct publication client of CA1.  
 2. CA1 may support the registration of multiple publishers by CA2, by using the publisher_request/repository_response XML exchange defined in RFC 8183.  CA2 would then be able to register a separate publisher on behalf of CA3.
 3. CA2 may operate a publication proxy service (per e.g. [@rpki-publication-proxy]), which acts as the publication server for CA3.  This proxy would set aside part of CA2's namespace at CA1 for the publication of CA3's objects, adjusting and forwarding requests from CA3 to CA1 accordingly.
 
For options 1 and 2, CAs operating as CA1 should consider the implications of providing direct publication service to CA3 in this way: for example, CA3 may expect publication service technical support from CA1 directly.
```